src/Analyzers/CSharp/Analyzers/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,111 @@
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class ConditionalExpressionPlacementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public ConditionalExpressionPlacementDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.ConditionalExpressionPlacementDiagnosticId,
+                   EnforceOnBuildValues.ConsecutiveBracePlacement,
+                   CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression,
+                   new LocalizableResourceString(
+                       nameof(CSharpAnalyzersResources.Blank_line_not_allowed_after_conditional_expression_token), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxTreeAction(AnalyzeTree);
+
+        private void AnalyzeTree(SyntaxTreeAnalysisContext context)
+        {
+            var option = context.GetCSharpAnalyzerOptions().AllowBlankLineAfterTokenInConditionalExpression;
+            if (option.Value)
+                return;
+
+            Recurse(context, option.Notification.Severity, context.Tree.GetRoot(context.CancellationToken));
+        }
+
+        private void Recurse(SyntaxTreeAnalysisContext context, ReportDiagnostic severity, SyntaxNode node)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            // Don't bother analyzing nodes that have syntax errors in them.
+            if (node.ContainsDiagnostics)
+                return;
+
+            if (node is ConditionalExpressionSyntax conditionalExpression)
+                ProcessConstructorInitializer(context, severity, conditionalExpression);
+
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.IsNode)
+                    Recurse(context, severity, child.AsNode()!);
+            }
+        }
+
+        private void ProcessConstructorInitializer(
+            SyntaxTreeAnalysisContext context, ReportDiagnostic severity, ConditionalExpressionSyntax conditionalExpression)
+        {
+            // Only if both tokens are not ok do we report an error.  For example, the following is legal:
+            //
+            //  var x =
+            //      goo ? bar :
+            //      baz ? quux : ztesh;
+            //
+            // despite one colon being at the end of the line.
+            if (IsOk(conditionalExpression.QuestionToken) ||
+                IsOk(conditionalExpression.ColonToken))
+            {
+                return;
+            }
+
+            context.ReportDiagnostic(DiagnosticHelper.Create(
+                this.Descriptor,
+                conditionalExpression.QuestionToken.GetLocation(),
```

Comment:
The location might not be correct.

Reply:
how so?

Reply:
@CyrusNajmabadi Isn't it sometimes on the `:` and sometimes on the `?` ? (depending on which `IsOk` call that failed)

Reply:
Both have to be not-ok for us to report the issue.  So teh question token will always be not ok.

Reply:
Ah yes, my bad.